code/modules/paperwork/paperplane.dm

```diff
@@ -0,0 +1,130 @@
+// Ported from TG
+/obj/item/weapon/paperplane
+	name = "paper plane"
+	desc = "Paper, folded in the shape of a plane."
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "paperplane"
+	throw_range = 7
+	throw_speed = 1
+	throwforce = 0
+	w_class = WEIGHT_CLASS_TINY
+	burn_state = FLAMMABLE
+	burntime = 5
+	no_spin = TRUE
+
+	var/obj/item/weapon/paper/internal_paper
+
+/obj/item/weapon/paperplane/New(loc, obj/item/weapon/paper/new_paper)
+	. = ..()
+	pixel_y = rand(-8, 8)
+	pixel_x = rand(-9, 9)
+	if(new_paper)
+		internal_paper = new_paper
+		flags = new_paper.flags
+		color = new_paper.color
+		new_paper.forceMove(src)
+	else
+		internal_paper = new /obj/item/weapon/paper(src)
+	update_icon()
+
+/obj/item/weapon/paperplane/Destroy()
+	if(internal_paper)
+		qdel(internal_paper)
+		internal_paper = null
+	return ..()
+
+/obj/item/weapon/paperplane/suicide_act(mob/living/user)
+	user.Stun(10)
+	user.visible_message("<span class='suicide'>[user] jams [name] in \his nose. It looks like \he's trying to commit suicide!</span>")
+	user.EyeBlurry(6)
+	var/obj/item/organ/internal/eyes/E = user.get_int_organ(/obj/item/organ/internal/eyes)
+	if(E)
+		E.take_damage(8, 1)
+	sleep(10)
+	return BRUTELOSS
+
+/obj/item/weapon/paperplane/update_icon()
+	overlays.Cut()
+	var/list/stamped = internal_paper.stamped
+	if(!stamped)
+		stamped = new
+	else if(stamped)
+		for(var/S in stamped)
+			var/obj/item/weapon/stamp = S
+			var/image/stampoverlay = image('icons/obj/bureaucracy.dmi', "paperplane_[initial(stamp.icon_state)]")
+			overlays += stampoverlay
+
+/obj/item/weapon/paperplane/attack_self(mob/user)
+	to_chat(user, "<span class='notice'>You unfold [src].</span>")
+	var/atom/movable/internal_paper_tmp = internal_paper
```

Comment:
why is this cast as a new var? Why not just 

```
	if(internal_paper)
		internal_paper.forceMove(loc)
		internal_paper = null
		qdel(src)
		user.put_in_hands(internal_paper)
```

Reply:
That'd delete the paper before it is put in your hand. I learned the hard way.

Reply:
It...shouldn't, but uh. Try putting the `qdel(src)` last, in that case.